tornado/locks.py

```diff
@@ -0,0 +1,62 @@
+__all__ = ['Condition']
+
+import collections
+
+from tornado import gen, ioloop
+from tornado.concurrent import Future
+
+
+class Condition(object):
+    """A condition allows one or more coroutines to wait until notified.
+
+    Like a standard Condition_, but does not need an underlying lock that
+    is acquired and released.
+
+    .. _Condition: http://docs.python.org/library/threading.html#threading.Condition
+    """
+
+    def __init__(self, io_loop=None):
```

Comment:
In 4.1 I deprecated `io_loop` arguments in favor of just relying on `IOLoop.current()` everywhere. (We can't always pass an `io_loop` around, e.g. into the coroutine decorator). We shouldn't add them in new interfaces.